core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -4042,6 +3995,17 @@ object Stream extends StreamLowPriority {
       go(init, self)
     }
 
+    /**
+      * Takes the first value output by this stream and returns it in the result of a pull.
+      * If no value is output before the stream terminates, the pull is failed with a `NoSuchElementException`.
+      * If more than 1 value is output, everything beyond the first is ignored.
+      */
```

Comment:
I wonder if there is any particular reason for the inconsistency with `compile.lastOrError`

Reply:
What's the inconsistency, just the name? I'll rename to `headOrError`.

Reply:
Just the `last` vs `first` behaviour. I can see how `Pull` is more `first`-y (`uncons`), but it's inconsistent with `lastOrError`. Not a big deal though